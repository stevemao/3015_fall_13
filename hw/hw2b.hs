

-- You need to write the functions curry3 and uncurry3 and add them to
-- this file.

curry f x y = f (x,y)
uncurry f (x,y) = f x y

plus1 x y  = x + y 
plus2 (x, y)  = x + y 

-- See http://en.wikipedia.org/wiki/Discriminant for a discussion of
-- the discriminant of a quadratic polynomial (a polynomial of the
-- form ax^2 + bx + c = 0.)  The delta function below computes the
-- discriminant of a quadratic polynomial.

delta (a,b,c) = sqrt (b ^ 2 - 4 * a * c)


