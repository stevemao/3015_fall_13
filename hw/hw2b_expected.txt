Prelude> :l hw2b.hs
:l hw2b.hs
[1 of 1] Compiling Main             ( hw2b.hs, interpreted )
Ok, modules loaded: Main.
*Main> :t curry3
:t curry3
curry3 :: ((t, t1, t2) -> t3) -> t -> t1 -> t2 -> t3
*Main> :t uncurry3
:t uncurry3
uncurry3 :: (t -> t1 -> t2 -> t3) -> (t, t1, t2) -> t3
*Main> :t delta
:t delta
delta :: (Floating t) => (t, t, t) -> t
*Main> :t curry3 delta
:t curry3 delta
curry3 delta :: (Floating t) => t -> t -> t -> t
*Main> curry3 delta 4.2 7 2.3
curry3 delta 4.2 7 2.3
3.2186953878862163
*Main> uncurry3 (curry3 delta) (4.2, 7, 2.3)
uncurry3 (curry3 delta) (4.2, 7, 2.3)
3.2186953878862163
